Index: PhysicsTools/PatAlgos/python/tools/helpers.py
===================================================================
RCS file: /local/reps/CMSSW/CMSSW/PhysicsTools/PatAlgos/python/tools/helpers.py,v
retrieving revision 1.19
diff -u -p -r1.19 helpers.py
--- PhysicsTools/PatAlgos/python/tools/helpers.py	25 Oct 2011 14:53:10 -0000	1.19
+++ PhysicsTools/PatAlgos/python/tools/helpers.py	7 Oct 2013 06:29:09 -0000
@@ -58,7 +58,7 @@ class MassSearchReplaceAnyInputTagVisito
         self._moduleLabelOnly=moduleLabelOnly
     def doIt(self,pset,base):
         if isinstance(pset, cms._Parameterizable):
-            for name in pset.parameters_().keys():
+            for name in pset.parameterNames_():
                 # if I use pset.parameters_().items() I get copies of the parameter values
                 # so I can't modify the nested pset
                 value = getattr(pset,name) 
