Index: SimG4Core/Application/plugins/OscarProducer.cc
===================================================================
RCS file: /cvs_server/repositories/CMSSW/CMSSW/SimG4Core/Application/plugins/OscarProducer.cc,v
retrieving revision 1.12
diff -u -p -r1.12 OscarProducer.cc
--- SimG4Core/Application/plugins/OscarProducer.cc	24 May 2009 14:43:30 -0000	1.12
+++ SimG4Core/Application/plugins/OscarProducer.cc	17 Nov 2010 07:57:12 -0000
@@ -120,7 +120,11 @@ OscarProducer::~OscarProducer() 
 
 void OscarProducer::beginRun(edm::Run & r, const edm::EventSetup & es)
 {
-  m_runManager->initG4(es);
+  static bool runOnce = true;
+  if(runOnce) {
+    m_runManager->initG4(es);
+    runOnce = false;
+  }
 }
 
 
