Index: RecoLuminosity/LumiDB/python/cacheconfigParser.py
===================================================================
RCS file: /cvs_server/repositories/CMSSW/CMSSW/RecoLuminosity/LumiDB/python/cacheconfigParser.py,v
retrieving revision 1.4
diff -u -p -u -r1.4 cacheconfigParser.py
--- RecoLuminosity/LumiDB/python/cacheconfigParser.py	1 Sep 2010 15:34:09 -0000	1.4
+++ RecoLuminosity/LumiDB/python/cacheconfigParser.py	17 Feb 2011 06:54:14 -0000
@@ -11,7 +11,7 @@ class cacheconfigParser(object):
         nodelist=dom.getElementsByTagName('frontier-connect')[0].childNodes
         #print nodelist
         for node in nodelist:
-            if node.nodeType!=Node.TEXT_NODE:
+            if node.nodeType!=Node.TEXT_NODE and node.nodeType != Node.COMMENT_NODE:
                 tagname=node.tagName
                 attrs=node.attributes
                 for attrName in attrs.keys():
