Index: PhysicsTools/PFCandProducer/python/pfTaus_cff.py
===================================================================
RCS file: /cvs/CMSSW/CMSSW/PhysicsTools/PFCandProducer/python/Attic/pfTaus_cff.py,v
retrieving revision 1.19.2.1
diff -u -r1.19.2.1 pfTaus_cff.py
--- PhysicsTools/PFCandProducer/python/pfTaus_cff.py	21 Mar 2011 19:41:07 -0000	1.19.2.1
+++ PhysicsTools/PFCandProducer/python/pfTaus_cff.py	1 Apr 2011 11:23:04 -0000
@@ -62,15 +62,30 @@
 # is it correct collection w/o good leptons
 pfTausProducer.builders[0].pfCandSrc = pfJets.src
 
+pfTauPileUpVertices = cms.EDFilter(
+    "RecoTauPileUpVertexSelector",
+    src = cms.InputTag("offlinePrimaryVertices"),
+    minTrackSumPt = cms.double(5),
+    filter = cms.bool(False),
+)
+
 # PiZeroProducers
 pfJetsPiZeros = ak5PFJetsRecoTauPiZeros.clone()
 pfJetsLegacyTaNCPiZeros = ak5PFJetsLegacyTaNCPiZeros.clone()
 pfJetsLegacyHPSPiZeros = ak5PFJetsLegacyHPSPiZeros.clone()
 
-pfJetsPiZeros.src = pfJetTracksAssociatorAtVertex.jets
-pfJetsLegacyTaNCPiZeros.src = pfJetTracksAssociatorAtVertex.jets
-pfJetsLegacyHPSPiZeros.src = pfJetTracksAssociatorAtVertex.jets
-
+pfJetsPiZeros.jetSrc = pfJetTracksAssociatorAtVertex.jets
+pfJetsLegacyTaNCPiZeros.jetSrc = pfJetTracksAssociatorAtVertex.jets
+pfJetsLegacyHPSPiZeros.jetSrc = pfJetTracksAssociatorAtVertex.jets
+
+pfTauPFJets08Region = recoTauAK5PFJets08Region.clone()
+pfTauPFJets08Region.src = pfJetTracksAssociatorAtVertex.jets
+pfTauPFJets08Region.pfSrc = pfJets.src
+pfJetsPiZeros.jetRegionSrc = 'pfTauPFJets08Region'
+pfJetsLegacyTaNCPiZeros.jetRegionSrc = 'pfTauPFJets08Region'
+pfJetsLegacyHPSPiZeros.jetRegionSrc = 'pfTauPFJets08Region'
+pfTausProducer.piZeroSrc = "pfJetsLegacyTaNCPiZeros"
+pfTausProducer.jetRegionSrc = pfJetsLegacyTaNCPiZeros.jetRegionSrc
 
 pfTauTagInfoProducer = pfRecoTauTagInfoProducer.clone()
 pfTauTagInfoProducer.PFCandidateProducer = pfJets.src
@@ -84,6 +99,8 @@
 
 pfTausPreSequence = cms.Sequence(
     pfJetTracksAssociatorAtVertex + 
+    pfTauPFJets08Region +
+    pfTauPileUpVertices +
     pfTauTagInfoProducer +
     pfJetsPiZeros +
     pfJetsLegacyTaNCPiZeros +
